### DISCUSSION TITLE
How to Enable HTTPS

### DISCUSSION THREAD

**Discussion by MSoliven** [microrealestate/microrealestate#179] [General]
Has anyone successfully deployed this application with HTTPS enabled? I've created a LetsEncrypt SSL certificate using Certbot, but I don't know how the web server can reference it. I don't see any setting in the .env files for this. Thank you.

https://flaviocopes.com/express-letsencrypt-ssl/#generate-the-ssl-certificate-using-certbot

**Comment by MSoliven** [comment]
To answer my own question, I ended up installing NGINX reverse proxy (outside Docker) and implementing the Https protocol there instead of doing it inside Node Js. There's no point in encrypting the communication between the reverse proxy and the application because it's in the same machine anyway (a potential attacker could access the private key).

**Reply by camelaissani** [reply]
@MSoliven Same on my side. I use nginx as a reverse proxy. Thank you for having shared your solution.

**Comment by nmbgeek** [comment]
https://github.com/traefik/traefik or https://github.com/NginxProxyManager/nginx-proxy-manager are the simplest solutions

**Comment by camelaissani** [comment]
The new install procedure rely on Caddy and provides the https support out of the box.

https://github.com/microrealestate/microrealestate?tab=readme-ov-file#self-host-the-application
